### ISSUE TITLE
Bug: Generating invalid code in composed schema contexts

### ISSUE THREAD

**Issue by jabrandes** [acacode/swagger-typescript-api#51]
An invalid Typescript type is generated in contexts where all of the following criteria are met:
- The schema is a composed schema with an `allOf` specification
- The `allOf` contains at least two type specifications
- One of the `allOf` types contains an array property of `enum` values
- That array contains an enum value with a hyphen (`-`)

I suspect the issue to be the following code snippet:
https://github.com/acacode/swagger-typescript-api/blob/acc770b5b12a00e7c158d13a8194d196563067ab/src/typeFormatters.js#L31-L37
Since `checkAndRenameModelName` will find a hyphen in the described situation, Lodash's `startCase` function and whitespace removal is executed on the whole subtype, causing distortions in the output as shown below (undesired casing, missing punctuation and missing spacing). It might be a sufficient fix to just remove the `checkAndRenameModelName` there.

#### Swagger specification for reproduction
```yaml
openapi: 3.0.1
info:
  title: Test
  version: test
paths: {}
components:
  schemas:
    Test:
      type: object
      allOf:
        - type: object
          properties:
            x:
              type: array
              items:
                type: string
                enum:
                  - A-B
        - type: object
          properties:
            y:
              type: string
```

#### Generated code
```typescript
export type Test = XAB & { y?: string };
```

**Comment by js2me** [comment]
Hello @jabrandes !  
Thanks for this issue, it will be fixed at the next release!

**Comment by js2me** [comment]
@jabrandes the generated type should look like ?  
```ts
export type Test = { x: "A-B" } & { y?: string };
```

**Comment by jabrandes** [comment]
Like
```​export​ ​type​ ​Test​ ​=​ ​{​ ​x?: (​"A-B"​ ​)[]}​ & ​{​ ​y​?: ​string​ ​}​;```
I'm not sure if I got the nullability of `x` right, but `x` shoild be an array/subset of enum values.

**Comment by js2me** [comment]
It should be fixed in release 1.8.3 🙂
